src/test/java/com/artipie/rpm/http/RpmSliceDownloadITCase.java

```diff
@@ -0,0 +1,154 @@
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+final class RpmSliceDownloadITCase {
+
+    /**
+     * Vertx instance.
+     */
+    private static final Vertx VERTX = Vertx.vertx();
+
+    /**
+     * Vertx slice server instance.
+     */
+    private VertxSliceServer server;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Port.
+     */
+    private int port;
+
+    @Test
+    void installs() throws Exception {
+        final TestRpm rpm = new TestRpm.Time();
+        this.start(rpm, Permissions.FREE, Identities.ANONYMOUS);
+        MatcherAssert.assertThat(
+            this.executeCommand(
+                String.format(
+                    "http://host.testcontainers.internal:%d/%s.rpm",
+                    this.port, rpm.name()
+                )
+            ),
+            new StringContainsInOrder(new ListOf<>(rpm.name(), "Complete!"))
+        );
+    }
+
+    @Test
+    void installsWithAuth() throws Exception {
+        final String john = "john";
+        final String pswd = "123";
+        final TestRpm rpm = new TestRpm.Time();
+        this.start(
+            rpm,
+            (name, perm) -> john.equals(name) && "download".equals(perm),
+            new BasicIdentities(
+                (name, pass) -> {
+                    final Optional<String> res;
+                    if (john.equals(name) && pswd.equals(pass)) {
+                        res = Optional.of(name);
+                    } else {
+                        res = Optional.empty();
+                    }
+                    return res;
+                }
+            )
+        );
+        MatcherAssert.assertThat(
+            this.executeCommand(
+                String.format(
+                    "http://%s:%s@host.testcontainers.internal:%d/%s.rpm", john,
+                    pswd, this.port, rpm.name()
+                )
+            ),
+            new StringContainsInOrder(new ListOf<>(rpm.name(), "Complete!"))
+        );
+    }
+
+    @AfterEach
+    void stopContainer() {
+        this.server.close();
+        this.cntn.stop();
+    }
+
+    @AfterAll
+    static void close() {
+        RpmSliceDownloadITCase.VERTX.close();
+    }
+
+    private String executeCommand(final String url) throws IOException, InterruptedException {
+        return this.cntn.execInContainer(
+            "yum", "-y", "install", url
+        ).getStdout();
+    }
+
+    private void start(
+        final TestRpm rpm, final Permissions perms, final Identities auth
+    ) throws Exception {
+        final Storage storage = new InMemoryStorage();
+        rpm.put(storage);
+        this.server = new VertxSliceServer(
+            RpmSliceDownloadITCase.VERTX,
+            new LoggingSlice(new RpmSlice(storage, perms, auth, new RepoConfig.Simple()))
+        );
+        this.port = this.server.start();
+        Testcontainers.exposeHostPorts(this.port);
+        this.cntn = new GenericContainer<>("centos:latest")
```

Comment:
@olenagerasimova I'd use concrete version of Docker image to avoid unpredictable behavior when `latest` image updated, e.g. it can be `centos:centos8`